src/Microsoft.Azure.ServiceBus/ServiceBusConnectionStringBuilder.cs

```diff
@@ -14,12 +15,20 @@ public class ServiceBusConnectionStringBuilder
         const char KeyValueSeparator = '=';
         const char KeyValuePairDelimiter = ';';
         static readonly string EndpointScheme = "amqps";
-        static readonly string EndpointFormat = EndpointScheme + "://{0}.servicebus.windows.net";
         static readonly string EndpointConfigName = "Endpoint";
         static readonly string SharedAccessKeyNameConfigName = "SharedAccessKeyName";
         static readonly string SharedAccessKeyConfigName = "SharedAccessKey";
         static readonly string EntityPathConfigName = "EntityPath";
 
+        string entityPath, sasKeyName, sasKey, endpoint;
+
+        /// <summary>
+        /// Instantiates a new <see cref="ServiceBusConnectionStringBuilder"/>
+        /// </summary>
+        public ServiceBusConnectionStringBuilder()
```

Comment:
@SeanFeldman , we have left it public for the sake of flexibility. Right now we are forced to use the constructor which accepts sasKeyName and sasKeyValue (if you don't have the entire connection string). In the near future, we will have to expose other types of authentication including sasTokens. Restricting to use a `builder` unless you have all the necessary variables is no longer a `builder` and strictly becomes a formatter. You lose the flexibility of what a builder is supposed to do.